apps/dashboard/app/apps/ood_app.rb

```diff
@@ -264,7 +264,8 @@ def run_setup_production
           #
           # This makes the execution of the setup-production script use the same ruby versions
           # that Passenger uses when launching the app.
-          cmd = File.exist?('./Gemfile') ? "bundle exec #{setup}" : setup
+          cmd = [setup]
+          cmd.prepend(%w[bundle exec]) if File.exist?('./Gemfile')
```

Comment:
I've got to actually test this now, I don't know if we need to splat this array or not.

Reply:
To test it in dev mode, you can make this hack on the controller. Navigate to `My sandbox apps` and try to `Launch Ood Dashboard` and it'll go through this code.

```diff
diff --git a/apps/dashboard/app/controllers/apps_controller.rb b/apps/dashboard/app/controllers/apps_controller.rb
index b8373bd4..4712b409 100644
--- a/apps/dashboard/app/controllers/apps_controller.rb
+++ b/apps/dashboard/app/controllers/apps_controller.rb
@@ -21,7 +21,7 @@ class AppsController < ApplicationController
     #FIXME: the only thing about this action that feels wrong
     #is it is a GET and we are doing a setup (changing something) in response to
     #this request
-    @app.run_setup_production unless @app.type == :dev
+    @app.run_setup_production #unless @app.type == :dev
 
     app_url = @app.url
```